lib/Backend/GlobOpt.cpp

```diff
@@ -19058,6 +19058,44 @@ GlobOpt::OptDstIsInvariant(IR::RegOpnd *dst)
 }
 
 void
+GlobOpt::OptHoistToLandingPadUpdateValueType(
+    BasicBlock* landingPad,
+    IR::Instr* instr,
+    IR::Opnd* opnd,
+    Value* opndVal)
+{
+    if (instr->m_opcode == Js::OpCode::FromVar)
```

Comment:
Updating type for `FromVar` causes some problem in lowerer because it has some type restrictions, such as the source cannot be a definite type. This could be violated if we update it to landingpad. Better leave it to the code in `OptHoistInvariant` to handle it.

---
In reply to: [101556159](https://github.com/Microsoft/ChakraCore/pull/2538#discussion_r101556159) [](ancestors = 101556159)